### ISSUE TITLE
Hard breaks aren't reflected in output

### ISSUE THREAD

**Issue by amartini51** [apple/swift-docc#477]
### Description

The [CommonMark spec](https://spec.commonmark.org/0.30/#hard-line-break) allows writers to specify a line break (not a paragraph break) by writing two or more spaces at the end of the line, or by writing a backslash (`\`) at the end of the line.  Currently, the rendered content from DocC doesn't break the line at that location.

Example input:

```
This text should\
appear on two lines.

This is the second paragraph.
```

### Checklist

- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/apple/swift-docc/issues).

rdar://101684224

**Comment by gwynne** [comment]
Is there any hope of some motion on this? Still hasn't been fixed as of the Xcode 15 beta/Swift 5.9

**Comment by Kyle-Ye** [comment]
> Is there any hope of some motion on this? Still hasn't been fixed as of the Xcode 15 beta/Swift 5.9

Sorry, I almost forgot the issue 😢. Thanks for the reminding.

**Comment by Kyle-Ye** [comment]
> Is there any hope of some motion on this? Still hasn't been fixed as of the Xcode 15 beta/Swift 5.9

I've added the swift-docc PR via #633 and created the corresponding issue to swift-docc-render via https://github.com/apple/swift-docc-render/issues/697. When both of them are ready, the issue will be closed.

Then we may consider cherry-pick this back to the release/5.9 branch.

**Comment by Kyle-Ye** [comment]
> Is there any hope of some motion on this? Still hasn't been fixed as of the Xcode 15 beta/Swift 5.9

The fix is landed on main branch of swift-docc and swift-docc-render. And we'll cherry-pick it back to release/5.9.

**Comment by mportiz08** [comment]
Both PRs have been merged and also cherry-picked to release/5.9 now :)
